docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_8.md

```diff
@@ -0,0 +1,156 @@
+# ExperimentalDetectronGenerateProposalsSingleImage {#openvino_docs_ops_detection_ExperimentalDetectronGenerateProposalsSingleImage_8}
+
+**Versioned name**: *ExperimentalDetectronGenerateProposalsSingleImage-8*
+
+**Category**: *Object detection*
+
+**Short description**: The *ExperimentalDetectronGenerateProposalsSingleImage* operation computes ROIs and their scores
+based on input data.
+
+**Detailed description**: The operation performs the following steps:
```

Comment:
1. Is the new version of ExperimentalDetectronGenerateProposalsSingleImage a generalization of the current version, in the sense of an interface?

1. Is the new version of ExperimentalDetectronGenerateProposalsSingleImage a generalization of the current version, in the sense of an algorithm of computations?

1. Will we have backward compatibility in results of the current and the new versions of the operation and how should be values of new parameters for such compatibility?

Could you write please all it, as a short comparison of the current and a new versions of the operation in the section `Details` of the PR?

Reply:
Could you also write please the purpose of these changes, in the same section `Details`?

Reply:
Hi @vgavrilo , I have updated the PR. The purpose is to support the paddle generate proposals v2 https://github.com/PaddlePaddle/PaddleDetection/blob/3c78716d30945e89129c1839453932482f4a6974/ppdet/modeling/ops.py#L1424 . The reference design to support the paddle generate proposals is https://github.com/openvinotoolkit/openvino/pull/10221 .
1. It is not a generalization of the current version in the sense of an interface. The proposals output becomes dynamic. and add another output port to indicate the number of proposals.
2. It is not a generalization of the current version in the sense of an algorithm.
    2.1 put roi size filter after the partial sort. Because Paddle and pytorch generate proposals both put the size filter after the partial sort. (https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/detection/generate_proposals_v2_op.cc and https://github.com/pytorch/pytorch/blob/master/caffe2/operators/generate_proposals_op.cc). Do you think whether we should follow the algorithm of paddle and pytorch?
    2.2 coordinate offset changes on https://github.com/meiyang-intel/openvino/blob/06572611074563e9959dba5e36a61fd7bff6a64a/src/plugins/intel_cpu/src/nodes/mkldnn_experimental_detectron_generate_proposals_single_image_node.cpp#L435 . After the change the result is the same as paddle result. Other wise there is precision issue. Is it a bug?
3. the changes just mentioned is not compatible with current version. I tend to apply the change regarding 1 and 2.2 . What do you think? About 2.1 do you have any comment?

Reply:
I think that because your changes are not compatible with the current version of the operation `ExperimentalDetectronGenerateProposalsSingleImage`, you should give other name of your operation, and you shouldn't create new version of an existing one.

Next, I think that the sentence 
> Please refer to CVS-71058 for more details of paddlepaddle generate_proposals op.

is very strange, because this sentence contains a link to an internal Intel ticket, although OpenVINO is an open source project.

Consequently, could you please write out short analysis of the PaddlePaddle operation, to justify the need for new OpenVINO operation with semantics as in this PR? In this analysis, could you please explain why cannot the operation proposed by you be decomposed into more simple operations? This analysis should be written in this PR `Description` field, and without any links to internal resources.

As for more detailed answer, I need to study content at links mentioned by you.

Also, could you please give me a link to the PaddlePadle specification of the proposed operation, and write out this link in the field `Description`?

Reply:
I don't find the specification of paddle generate proposal v2. You can refer to https://github.com/PaddlePaddle/PaddleDetection/blob/3c78716d30945e89129c1839453932482f4a6974/ppdet/modeling/ops.py#L1424 and https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/detection/generate_proposals_v2_op.cc .

Reply:
It's a pity not using the existing operation in openvino because they are nearly the same operation.
Regarding 2.1 issue in https://github.com/openvinotoolkit/openvino/pull/9788#discussion_r802391638 , I glimpse the detectron2 generate proposal and seems it use pytorch generate proposal https://github.com/pytorch/pytorch/blob/master/caffe2/operators/generate_proposals_op.cc and it's the same as the paddle. Please correct me if I'm wrong. If it is true, I think it is ok to change the existing operation.
Regarding 2.2, is it a bug?

Reply:
Could you please replace the sentence 
`Please refer to CVS-71058 for more details of paddlepaddle generate_proposals op.
`
with the analysis from the ticket CVS-71058, include links? OpenVINO is open source project, and, consequently, such sentences are very strange.

Reply:
have deleted and put it in tickets

Reply:
First of all, initially all operations `ExperimentalDetectron*` were implemented as experimental operations, and only in CPU and VPU. Some time ago, were written specifications of all these operations (including `ExperimentalDetectronGenerateProposalsSingleImage`). All these specifications were written according to CPU plugin implementation. And more, there are ONNX models relying semantics as the current implementation in CPU plugin.

Because you say that 
`the changes just mentioned is not compatible with current version`
and in view of all abovementioned about operations `ExperimentalDetectron*` history, **you shouldn't create new version of an existing operation, but you should create new operation with the different name**.

Reply:
That's not an old version. That is my branch and I changed it in order to get the same result from paddle. Can you tell me why is it 1.0 not coordinates_offset?
Whether it is compatible or not should be discussed and it depends on the issue I mentioned below:
1. why https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mkldnn_experimental_detectron_generate_proposals_single_image_node.cpp#L408 it is 1.0 and not coordinates_offset?
2. why in experimental detectron generate proposal the ROI size filter is before partial sort? I have this question because in pytorch, paddle and detectron2, ROI size filter is all after partial sort.
These 2 questions are the very key points for compatible issue.

Reply:
Yes, the corresponding argument of the `refine_anchors()` is called `coordinates_offset`. But 1) where the variable `coordinate_offset` is defined in the function `execute()` or in the class of this MKLDNN operation?; 2) why should we use a separate variable, if we use it only once?